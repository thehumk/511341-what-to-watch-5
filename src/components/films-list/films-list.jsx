import React from 'react';
import PropTypes from 'prop-types';
import {propsForFilms} from '../../utils/prop-types';
import FilmCard from '../film-card/film-card';
import ShowMoreButton from '../show-more-button/show-more-button';
import {withFilmsList} from '../../hocs/with-films-list/with-films-list';

const FilmsList = (props) => {
  const {films, quantityRenderFilms, cardHoverHandler, cardLeaveHoverHandler, showMoreButtonClickHandler} = props;
  return (
    <>
      <div className="catalog__movies-list">
        {films.slice(0, quantityRenderFilms).map((elem) => (
          <FilmCard
            key={elem.id}
            film={elem}
            cardHoverHandler={cardHoverHandler}
            cardLeaveHoverHandler={cardLeaveHoverHandler}
          />
        ))}
      </div>
      <div className="catalog__more">
        {quantityRenderFilms < films.length ? <ShowMoreButton clickHandler={showMoreButtonClickHandler}/> : ``}
      </div>
    </>
  );
};

FilmsList.propTypes = {
  films: PropTypes.arrayOf(propsForFilms).isRequired,
  quantityRenderFilms: PropTypes.number.isRequired,
  cardHoverHandler: PropTypes.func.isRequired,
  cardLeaveHoverHandler: PropTypes.func.isRequired,
  showMoreButtonClickHandler: PropTypes.func.isRequired,
};

export {FilmsList};
export default withFilmsList(FilmsList);
