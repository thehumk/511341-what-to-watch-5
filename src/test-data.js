export const noop = () => {};

export const MATCH = {params: {id: `1`}};

export const mockComments = [
  {
    comment: `123`,
    date: `2020-10-29T13:38:44.769Z`,
    id: 1,
    rating: 4.6,
    user: {
      id: 12,
      name: `Isaac`
    },
  },
  {
    comment: `1234`,
    date: `2020-10-29T13:38:45.769Z`,
    id: 2,
    rating: 4.5,
    user: {
      id: 13,
      name: `Isa`
    },
  }
];

/* eslint-disable */

export const mockFilms = [
  {
    background_color: `#73B39A`,
    background_image: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
    description: `A young man who was sentenced to seven years in prison for robbing a post office ends up spending three decades in solitary confinement. During this time, his own personality is supplanted by his alter-ego, Charles Bronson.`,
    director: `Nicolas Winding Refn`,
    genre: `Action`,
    id: 1,
    is_favorite: true,
    name: `Bronson`,
    poster_image: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
    preview_image: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
    preview_video_link: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    rating: 3.6,
    released: 2008,
    run_time: 92,
    scores_count: 109661,
    starring: [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
    video_link: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  },
  {
    background_color: `#73B39A`,
    background_image: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
    description: `A young man who was sentenced to seven years in prison for robbing a post office ends up spending three decades in solitary confinement. During this time, his own personality is supplanted by his alter-ego, Charles Bronson.`,
    director: `Nicolas Winding Refn`,
    genre: `Drama`,
    id: 2,
    is_favorite: false,
    name: `Bronson`,
    poster_image: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
    preview_image: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
    preview_video_link: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    rating: 3.6,
    released: 2008,
    run_time: 92,
    scores_count: 109661,
    starring: [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
    video_link: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  },
];

/* eslint-enable */

export const state = {
  DATA: {
    films: mockFilms,
    promoFilm: mockFilms[0],
    genresList: [`All`, `Action`, `Drama`],
    isApplicationReady: false,
  },
  FILMS_STATUS: {
    activeGenre: `All`,
    renderFilmsCount: 8,
  },
  USER: {
    authorizationStatus: `NO_AUTH`,
    user: null,
  },
  COMMENTS: {
    comments: mockComments,
  },
};
