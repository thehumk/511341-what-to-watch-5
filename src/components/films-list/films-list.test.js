import React from 'react';
import {BrowserRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';
import {FilmsList} from './films-list';
import {noop, mockFilms, state} from '../../test-data';

it(`Should FilmsList render correctly`, () => {
  const tree = renderer.create(
      <BrowserRouter>
        <FilmsList
          films={mockFilms}
          quantityRenderFilms={state.FILMS_STATUS.renderFilmsCount}
          cardHoverHandler={noop}
          cardLeaveHoverHandler={noop}
          showMoreButtonClickHandler={noop}
        />
      </BrowserRouter>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
