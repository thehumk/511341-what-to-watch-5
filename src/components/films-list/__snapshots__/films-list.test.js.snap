// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should FilmsList render correctly 1`] = `
Array [
  <div
    className="catalog__movies-list"
  >
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="/films/1"
        onClick={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <video
            height="175"
            muted={true}
            poster="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg"
            src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <span>
            Bronson
          </span>
        </h3>
      </a>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="/films/2"
        onClick={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <video
            height="175"
            muted={true}
            poster="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg"
            src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <span>
            Bronson
          </span>
        </h3>
      </a>
    </article>
  </div>,
  <div
    className="catalog__more"
  >
    
  </div>,
]
`;
